ci/scripts/exec-check.sh

```diff
@@ -5,32 +5,23 @@
 
 set -e
 
-allowed_suffixes=(
-    py
-    sh
+# Find executable files that should not be executable.
+bad_files=$(
+find . \
+    -name .git -prune -o \
+    -name vendor -prune -o \
+    -type f \
+    -executable \
+    -name '*.*' \
+    -not -name '*.py' \
+    -not -name '*.sh' \
+    -print
 )
 
-suff_re=""
-for suff in "${allowed_suffixes[@]}"; do
-    suff_re="${suff_re}$suff\\|"
-done
-suff_re="\\(${suff_re:0:-2}\\)"
-
-path_re=".*\\.${suff_re}$"
-
-TMPFILE="$(mktemp)" || {
-    echo >&2 "Failed to create temporary file"
-    exit 1
-}
-trap 'rm -f "$TMPFILE"' EXIT
-
-find -name vendor -prune -o \
-     -name .git -prune -o \
-     -type f -executable -name '*.*' -not -regex "$path_re" \
-     -print >"$TMPFILE"
-if [ -s "$TMPFILE" ]; then
+# Fail if any exist.
+if [ -s "$bad_files" ]; then
```

Comment:
`-s` expects a path. You probably want to use `-n`?

Reply:
Thanks, this was working when I tested with one file because `"$bad_files"` was the name of a file. It doesn't work with multiple files. Changed to `-n`.